server/eclipse-project/src/main/java/us/freeandfair/corla/model/AuditInfo.java

```diff
@@ -90,6 +90,14 @@
   private Map<String, Set<String>>
     canonicalContests = new TreeMap<String, Set<String>>();
 
+  /** map of contestName to choices **/
+  @Convert(converter = CountyCanonicalContestsMapConverter.class)
+  @Column(name = "canonical_choices", columnDefinition = "text")
+  // @MapKey(name = "my_id") // not sure this is needed
+  private Map<String, Set<String>>
+    canonicalChoices = new TreeMap<String, Set<String>>();
```

Comment:
I didn't see this on the first read-through, but this mapping doesn't preserve which county the contest belonged to - for instance, we wouldn't be able to update just a single county's poor choices for a multi-county contest.

I think we'll need to rename and augment the contest data structure to hold all three pieces of information: county, contest, and choices. Whether it's best nested or flat is up to you...

Reply:
I believe the canonicalChoices could replace the canonicalContests since contest names are unique - the canonical ones that is. The ones that are getting fixed are not, so we would need the county id on the post request to set the contest name, but not in the canonical list. Am I right about that?

Reply:
You might be, but I think we'd need to confirm whether choices *must, in all cases* be the same for a particular contest regardless of county, or whether they can differ for a particular contest across different counties. The CSV file format we're accepting allows for county-specific choices, the ACVR uploads would allow for this, and then the math only cares about contest discrepancies, so it seems like we could be imposing an artificial constraint here, unless that's what Colorado wants.